src/native/fs/fs_napi.cpp

```diff
@@ -563,30 +639,45 @@ struct FileRead : public FSWorker
     }
 };
 
-// struct FileWrite : public FSWorker
-// {
-//     FileWrap *_wrap;
-//     const uint8_t* _buf;
-//     size_t _len;
-//     FileWrite(const Napi::CallbackInfo& info) : FSWorker(info)
-//     {
-//         _wrap = FileWrap::Unwrap(info.This().As<Napi::Object>());
-//         _buf = info[0].As<Napi::Buffer<uint8_t>>();
-//         _buf = info[1].As<Napi::Value<size_t>>();
-//         // TODO get buffer from info[0]
-//     }
-//     virtual void Work()
-//     {
-//         int fd = _wrap->_fd;
-//         std::string path = _wrap->_path;
-//         if (fd < 0) {
-//             SetError(XSTR() << "FS::FileWrite::Execute: ERROR not opened " << path);
-//             return;
-//         }
-
-//         // TODO - read(fd, buf...)
-//     }
-// };
+
+struct FileWrite : public FSWorker
+{
+    FileWrap *_wrap;
+    const uint8_t* _buf;
+    size_t _len;
+    ssize_t _bw;
+    FileWrite(const Napi::CallbackInfo& info) 
+        : FSWorker(info)
+        , _buf(0)
+        , _len(0)
+    {
+        _wrap = FileWrap::Unwrap(info.This().As<Napi::Object>());
+        auto buf = info[1].As<Napi::Buffer<uint8_t>>();
+        _buf = buf.Data();
+        _len = buf.Length();
+    }
+    virtual void Work()
+    {
+        int fd = _wrap->_fd;
+        std::string path = _wrap->_path;
+        if (fd < 0) {
+            SetError(XSTR() << "FS::FileWrite::Execute: ERROR not opened " << path);
+            return;
+        }
+        // TODO: Switch to pwrite when needed
+        _bw = write(fd, _buf, _len);
+        if (_bw < 0) {
+            SetSyscallError();
+        } else if ((size_t)_bw != _len) {
+            SetError(XSTR() << "FS::FileWrite::Execute: partial write error " << DVAL(_bw) << DVAL(_len));
+        }
+    }
+    virtual void OnOK()
+    {
+        Napi::Env env = Env();
+        _deferred.Resolve(Napi::Number::New(env, _bw));
```

Comment:
Why do we return the number of bytes if this call will always fail unless the number is exactly the same as the buffer length? Seems like it would be a simpler API to just remove this return value.